### PR TITLE
Kira sierra club dc

### PR DESCRIPTION
**Pull request by kirakrishnan**
@csmcallister these changes extract the events from the sierra club DC website. I added some unit tests. I also tried to add the some unit tests which you added to casey_trees in the tests folder. In the test_fixtures file there is field `events_list` which has the current output. let me know if you want me to make any changes.

### CODE REVIEW

**Comment by csmcallister** [comment]
A few fixes:
- [ ] The `Event Cost` field needs to be a string of digit(s). You have: 
```python
'Sierra Club Potomac Region Outings (SCPRO) requests a '
'voluntary donation of $2 per person to help defray the '
'expenses of this all-volunteer program.'
```
Should be `'2'` for the case above. In another case, the event cost is described as a voluntary donation. To capture those cases as a "technically" free event, you could test if the string contains the substring `'voluntary'` or`' donation'` and, if so, set the cost to `'0'`.

Finally, could you add a test for this field with:
```python
    @httpretty.activate
    def test_events_schema_event_cost_type(self):
        '''
        Tests if the event cost is a string of digits
        '''
        httpretty.register_uri(method=httpretty.GET,
                                            uri=self.api,
                                            status=200,
                                            body=self.api_content)
        r = requests.get(self.api)
        content = r.content
        page = json.loads(content)
        events = handle_ans_page(page['eventList'])
        vals = []
        for event in events:
            for k in event:
                if k == 'Event Cost':
                    val = event[k]
                    vals.append(val)
        #empty strings are "falsy"
        result = all(x.isdigit() or not x for x in vals)
        self.assertTrue(result)
```
- [ ] I learned yesterday from Britta that the `Event Organizers` field should be the event source. So, for this, it would just be `Sierra Club` for each event. Also, I noticed that you named the directory `sierra_club_DC`, but the event list contains locations such as `"location":"8850 Gorman Rd, Laurel, MD 20723, USA"`. That API is pulling events for the whole area, which is great!

- [ ] Although latitude and longitude aren't required, those fields are present in the API data, so would you mind capturing those and then just commenting out that code. That'll help us when we want to develop the map!

- [ ] I moved all the tests to a directory in the root of the repository called `tests/`. That's also where CircleCI looks for the tests to run on a PR. Could you move your tests to a new file `tests/sierra_club_test.py`? There's a  `test/fixtures/` directory as well. And the testing utilities are in `tests/utils.py` so that you can just import them.
>Also, once you move your test file into that directory, add the following imports:
```python
import sys
from os import path
sys.path.append( path.dirname( path.dirname( path.abspath(__file__) ) ) )
```
>That'll let the unittest test runner run the tests from within the `test/` directory.

**Comment by kirakrishnan** [comment]
@csmcallister  The `Event Cost` field is different for some events. for example:
```
  'Event Cost': 'Price includes camping fee ($32 dollars a night), gasoline, '
                'food (restaurant available), and a volunteer $2 donation to '
                'the Sierra Club to help run the outings program.',

  'Event Cost': 'Prices range from $95 to $300 depending on your '
                'accommodations.',

  'Event Cost': '$6 mandatory fee for Brookside Nature Center and $2 voluntary '
                'donation to the Sierra Club',

  'Event Cost': 'A two dollar volunteer donation is requested to help run the '
                'Sierra Club PRO outings program. There is no charge on '
                'February 22 to visit Mount Vernon.',

  'Event Cost': 'There is a $7 recommended contribution for carpool drivers. '
                'If you are taking VA-267, there is a $3 toll per pass '
                'through. SCPRO requests a voluntary $2/person donation to '
                'support our many volunteer-led activities.',

```

**Comment by csmcallister** [comment]
For event cost descriptions with multiple amounts like that, you probably can't reliably scrape the correct amount. So I'd just submit an empty string, which would essentially leave that field undefined. Folks could then get the event cost from visiting the event's website.

To extract all the dollar amounts from those texts, you could use this [regular expression](https://pythex.org/?regex=(%3F%3A%5B%5C%24%5D%7B1%7D%5B%2C%5Cd%5D%2B.%3F%5Cd*)&test_string=%27There%20is%20a%20%247%20recommended%20contribution%20for%20carpool%20drivers.%20%27%0A%27If%20you%20are%20taking%20VA-267%2C%20there%20is%20a%20%243%20toll%20per%20pass%20%27%0Athrough.%20SCPRO%20requests%20a%20voluntary%20%242%2Fperson%20donation%20to%20%27%0A%27support%20our%20many%20volunteer-led%20activities.&ignorecase=0&multiline=0&dotall=0&verbose=0).

In python, something like the following should work:
```
import re
def get_event_cost(event_cost_description):
    event_cost_description_lower = event_cost_description.lower()
    currency_re = re.compile(r'(?:[\$]{1}[,\d]+.?\d*)')
    event_costs = re.findall(currency_re, event_cost_description)
    if len(event_cost) == 1 and "donation" not in event_cost_description_lower and "voluntary" not in event_cost_description_lower:
        event_cost = events_costs[0].split(".")[0].replace("$",'')
        event_cost = ''.join(s for s in event_cost if s.isdigit())
    else:
        event_cost = ''
    
    return event_cost
```

Basically, if there's just one dollar amount - and if the words `donation` and `voluntary` aren't in the event cost description - then we'll use that amount for the event cost. See how something like that works.

**Comment by kirakrishnan** [comment]
* The `Event Cost` field needs to be a string of digit(s). You have:
--added a function to handle this
* I learned yesterday from Britta that the `Event Organizers` field should be the event source. So, for this, it would just be `Sierra Club` for each event. Also, I noticed that you named the directory `sierra_club_DC`, but the event list contains locations such as `"location":"8850 Gorman Rd, Laurel, MD 20723, USA"`. That API is pulling events for the whole area, which is great!
--changed it to `Sierra Club'
* Although latitude and longitude aren't required, those fields are present in the API data, so would you mind capturing those and then just commenting out that code. That'll help us when we want to develop the map!
-- already captured and commented
* I moved all the tests to a directory in the root of the repository called `tests/`. That's also where CircleCI looks for the tests to run on a PR. Could you move your tests to a new file `tests/sierra_club_test.py`? There's a  `test/fixtures/` directory as well. And the testing utilities are in `tests/utils.py` so that you can just import them.
-- moved to tests to the directory. but dont no why it got conflicts. I commited some of my changes with wrong name. so I tried updating the name. It could be a reason I think. let me know your thoughts

**Comment by csmcallister** [comment]
I tried running the tests but got an import error:

```
(base) Charless-MacBook-Air:tests mcallistercs$ python3 -W ignore -m unittest sierra_club_test.py 
E
======================================================================
ERROR: sierra_club_test (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: sierra_club_test
Traceback (most recent call last):
  File "/Users/mcallistercs/anaconda/lib/python3.6/unittest/loader.py", line 153, in loadTestsFromName
    module = __import__(module_name)
  File "/Users/mcallistercs/Desktop/Github/capital-nature-ingest/tests/sierra_club_test.py", line 8, in <module>
    from lambdas.sierra_club_DC.lambda_function import handle_ans_page
ModuleNotFoundError: No module named 'lambdas'


----------------------------------------------------------------------
Ran 1 test in 0.000s

FAILED (errors=1)
```

I believe you need to update your imports in `sierra_club_test.py`

**Comment by kirakrishnan** [comment]
my bad scott. I updated path. let me know if it is still throwing errors

**Comment by csmcallister** [comment]
@kirakrishnan Sorry I didn't notice this earlier, but your event output is missing the `Event Venue Name`, which you can get from the `location` attribute for each event returned by the API. Also, the url for the API is still a global variable, so can you just tuck that into the function(s) that use it.

- [ ] `url` is no longer a global variable
- [ ] `Event Venue Name` added to each event output
- [ ] `test_events_schema_required_fields()` updated to include `Event Venue Name` in the `schema` set.

**Comment by kirakrishnan** [comment]
@csmcallister  currently these are all the fields present for each event. The location field has the address for venue but I dont see any venue name. 
image

values for location fields are like this:
```
"2000 Florida Ave NW, Washington, DC 20009, USA"
"120 Cherry St SE, Vienna, VA 22180, USA"
"977 Balls Hill Rd, McLean, VA 22101, USA"
"Sierra Club National Office at 50 F St. NW, Washington, DC"
50 F St NW #1100, Washington, DC 20001, USA"
```
And venue name was not bolded in the schema so I thought it is not a required field. I am thinking to split the location with "," and use the first value as venue name if venue name is required field? It might not be a prefect solution but I think it might be close. let me know your thoughts

**Comment by csmcallister** [comment]
Ah!! You’re right about the venue name not being bolded! That’s my bad. I’ll fix soon.

For now, I’d just use the location as-is. We can worry about parsing it better if we get feedback from the CapNature team.

**Comment by kirakrishnan** [comment]
using the entire location as `event venue name` for now. updated test case and test fixtures file. And I made the event venue name as a required field in the md file.
